FRW/Modules/Login/ViewModel/KeyStoreLoginViewModel.swift

```diff
@@ -0,0 +1,244 @@
+//
+//  KeyStoreLoginViewModel.swift
+//  FRW
+//
+//  Created by cat on 2024/8/19.
+//
+
+import Foundation
+import WalletCore
+import Web3Core
+
+import Flow
+import FlowWalletKit
+import SwiftUI
+
+// MARK: - KeyStoreLoginViewModel
+
+class KeyStoreLoginViewModel: ObservableObject {
+    // MARK: Internal
+
+    @Published
+    var json: String = ""
+    @Published
+    var password: String = ""
+    @Published
+    var wantedAddress: String = ""
+
+    @Published
+    var buttonState: VPrimaryButtonState = .disabled
+
+    var userName: String = ""
+
+    @Published
+    var wallet: FlowWalletKit.Wallet? = nil
+
+
+    @MainActor
+    func update(json _: String) {
+        update()
+    }
+
+    @MainActor
+    func update(password _: String) {
+        update()
+    }
+
+    func update(address _: String) {}
+
+    func onSumbit() {
+
+        UIApplication.shared.endEditing()
+        HUD.loading()
+        Task {
+            do {
+                let chainId = LocalUserDefaults.shared.flowNetwork.toFlowType()
+                privateKey = try PrivateKey.restore(
+                    json: json,
+                    password: password,
+                    storage: FlowWalletKit.PrivateKey.PKStorage
+                )
+                guard let privateKey = privateKey else {
```

Comment:
nit: in case of optional binding on a property, you can also write this as:

```suggestion
                guard let privateKey else {
```